### ISSUE TITLE
Additional configurability for MA0075

### ISSUE THREAD

**Issue by RenderMichael** [meziantou/Meziantou.Analyzer#1086]
I am interested in [MA0075](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0075.md), but only for a subset of types (in my case, `DateTime`).

Would it be possible to add a configuration knob to specify which types we are interested in (similar to `MA0003.expression_kinds`)

Sample
```ini
MA0075.included_types = System.DateTime, enum
```

**Comment by meziantou** [comment]
The rule can already opt-out types https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/CultureInsensitiveTypeAttribute.md#assembly-level-annotation-for-external-types

How many false-positive do you have with the default configuration?

**Comment by RenderMichael** [comment]
I have several hundred false-flags for various BCL types. I was hoping to avoid code changes as they are less portable for our configuration, and instead handle it all in the `.editorconfig`.

Type opt-out would be just as good as opt-in, given how narrow the list is. Would it be possible to add a configuration knob that's equivalent to `[CultureInsensitiveType]`, directly in the .editorconfig? You have more insight into how these things should be laid out, but I'm picturing something similar to the OP code snippet:

```ini
MA0075.exclude_types = int, float, enum
```

**Comment by meziantou** [comment]
editorconfig would be possible, but it's not as good for performance as you cannot use caching. Indeed, the configuration can be different per file, instead of per project. So, the build would be slower.

One way to handle it across a solution without an `editorconfig` is
- Create a .cs file with the attributes to opt-out types
- Use the Directory.Build.props to include it automatically to all projects `<Compile Include="$(MSBuildThisFileDirectory)file.cs" Visible="false" />` (Visible=false is optional avoid polluting the solution explorer)

Is this workaround good enough?

**Comment by RenderMichael** [comment]
All of this makes sense; I understand how crucial performance is for this analyzer suite. The existing workarounds are workable. Thanks for considering!
